### PR TITLE
Add HF CLI Upload to E2E Workflow tutorial

### PR DESCRIPTION
**Pull request by joecummings**
#### Context
- Per discussion with @Wauplin on #698, we should feature the `huggingface-cli upload` command for a true e2e workflow

#### Changelog
- Add section to E2E workflow tutorial
- Remove `hf_upload` directory

#### Test plan
- Eyes

Screenshot 2024-04-16 at 1 44 17 PM

### CODE REVIEW

File: docs/source/tutorials/e2e_flow.rst [hunk 1]
```diff
@@ -522,5 +522,31 @@ The output should look something like this:
 
 And thats it! Try your own prompt!
 
+Uploading your model to the Hugging Face Hub
+--------------------------------------------
+
+Your new model is working great and you want to share it with the world. The easiest way to do this
+is utilizing the ``huggingface-cli`` command, which is integrated with torchtune. Simply point the CLI
```
Comment (msaroufim) [inline docs/source/tutorials/e2e_flow.rst:529]: nit: I don't believe it's integrated anymore, its more like interops or works seamlessly with

File: torchtune/_cli/hf_upload/LICENSE [hunk 2]
```diff
@@ -1,125 +0,0 @@
-LLAMA 2 COMMUNITY LICENSE AGREEMENT
```
Comment (msaroufim) [inline torchtune/_cli/hf_upload/LICENSE:1]: so this was here on purpose but it's an artifact of when we only supported llama2

File: docs/source/tutorials/e2e_flow.rst [hunk 3]
```diff
@@ -522,5 +522,31 @@ The output should look something like this:
...
+to your finetuned model directory like so:
+
+.. code-block:: bash
+
+    huggingface-cli upload <hf-repo-id> <checkpoint-dir>
```
Comment (msaroufim) [inline docs/source/tutorials/e2e_flow.rst:534]: nit: mention where the checkpoint will be saved by default

**Review by msaroufim — APPROVED** [review]
LGTM but I'll let @matthewdzmura comment on what to do about the licensing change

File: docs/source/tutorials/e2e_flow.rst [hunk 4]
```diff
@@ -522,5 +522,31 @@ The output should look something like this:
...
+The command should output a link to your repository on the Hub. If the repository doesn't exist yet, it will
+be created automatically:
+
+.. code-block:: text
+
+    https://huggingface.co/<hf-repo-id>/tree/main/.
+
+.. note::
+
+    Before uploading, make sure you are authenticated with Hugging Face by running ``huggingface-cli login``.
```
Comment (Wauplin) [inline docs/source/tutorials/e2e_flow.rst:545]: FYI, You can also mention the "authentication" section from the quickstart guide (https://huggingface.co/docs/huggingface_hub/quick-start#authentication). And yeah, `huggingface-cli login` is perfectly valid!

**Review by Wauplin — APPROVED** [review]
Thanks @joecummings!

**Pull request merged**